### PR TITLE
feat: live component demo

### LINKED ISSUE #357: [Design] Component page theme/variant switcher — ready
As a user I want to be able to view the live demo component in the 4 different themes as well as be able to see the different variants and modifiers in one window instead of multiple.

### PR DESCRIPTION
**Pull request by vpicone**
Closes #357 

- [x] Mobile view
- [x] react-docgen for static class properties

### Editor/preview
- [x] Live component editor/preview
- [X] style editor
- [x] style preview
- [x] Carbon themes
- [x] truncate editor to allow full height knobs container
- [x] edit in codesandbox

~Initial code for all components~ spun off into #621

### Props panel
- [x] Props container style
- [x] State
- [x] Variant/Size
- [x] Sub-variant/Modifiers
- [x] Editable content
- [x] Custom prop types with docgen handler
- [x] Handle cases of one Component  or no props
	- component gets border bottom if solo

### CODE REVIEW

**Comment by vpicone** [comment]
@aagonzales The groups correspond to the component props which are pascal case (e.g hideLabel). Would we still want to include that info somewhere?

**Comment by aagonzales** [comment]
hmm it looks incorrect when there's only one word 🤷‍♀ What info are you referring to?

**Comment by vpicone** [comment]
It’s the actual thing that gets added/updated to the component when you change a knob. They’re case sensitive.

**Comment by vpicone** [comment]
@aagonzales if someone edits the code and there's an error, we have access to that info to display it: 
Screen Shot 2019-11-18 at 2 13 34 PM

Do you want to design something to display the error w/ an error notification or should we just 'eat' the error and not display anything?

**Comment by aagonzales** [comment]
Also just found this bug where if you scroll just a little further down the page the component window moves with the content. 
Nov-25-2019 12-08-09

**Comment by alisonjoseph** [comment]
Should we remove the "documentation" underneath each component window. This currently pulls in the readme files which relate to the vanilla versions of each component.

**Comment by vpicone** [comment]
@alisonjoseph yeah, I plan on removing the pointless H2 component name titles as well

**Comment by vpicone** [comment]
@alisonjoseph removed the excess titles. Content updates adding the demo component/removing titles from other components are going to be handled with #621

@aagonzales fixed the twerking jank

**Comment by vpicone** [comment]
@aagonzales fixed!

File: src/pages/components/accordion/code.mdx [hunk 1]
```diff
@@ -4,17 +4,13 @@ description: The accordion component delivers large amounts of content in a smal
...
+<ComponentDemo
+  knobs={{ AccordionItem: ['open'] }}
+  code={`
+<Accordion>
+  <AccordionItem title="title 1">content 1</AccordionItem>
+</Accordion>
+`}
+/>
 
 <ComponentDocs component="accordion"></ComponentDocs>
```
Comment (alisonjoseph) [inline src/pages/components/accordion/code.mdx]: I think we want to remove all of the `<ComponentDocs />` components, they pull in vanilla docs.

Comment (vpicone) [reply src/pages/components/accordion/code.mdx]: @alisonjoseph I opened https://github.com/carbon-design-system/carbon-website/pull/648 to take care of that along with replace the `ComponentCode` components.

**Comment by vpicone** [comment]
@alisonjoseph I'm gonna remove the `ComponentDocs` for the components affected here like you recommended, that way we can merge the other PR later this week after the rest of the components get the demo added.

**Pull request merged**